pydantic/_internal/_generate_schema.py

```diff
@@ -1289,13 +1293,20 @@ def set_discriminator(schema: CoreSchema) -> CoreSchema:
             schema = self._apply_discriminator_to_union(schema, field_info.discriminator)
             return schema
 
+        # Convert `@field_validator` decorators to `Before/After/Plain/WrapValidator` instances:
+        validators_from_decorators = []
+        for decorator in filter_field_decorator_info_by_field(decorators.field_validators.values(), name):
+            validators_from_decorators.append(_mode_to_validator[decorator.info.mode]._from_decorator(decorator))
+
         with self.field_name_stack.push(name):
             if field_info.discriminator is not None:
-                schema = self._apply_annotations(source_type, annotations, transform_inner_schema=set_discriminator)
+                schema = self._apply_annotations(
+                    source_type, annotations + validators_from_decorators, transform_inner_schema=set_discriminator
```

Comment:
This is an interesting pattern - I'm assuming, based on passing tests, that this preserves annotation order as expected.

Reply:
Indeed it does. Previously, annotations where applied and then the validators from decorators. Same happens now, as `_apply_annotations` will apply the `annotations` argument from "left to right".